Content.Server/Ghost/Roles/Components/GhostRoleMobSpawnerComponent.cs

```diff
@@ -18,9 +17,30 @@ public sealed class GhostRoleMobSpawnerComponent : Component
 
         [ViewVariables]
         public int CurrentTakeovers = 0;
+    }
 
+    /// <summary>
+    ///   Allows a ghost to take this role, spawning a new entity.
+    /// </summary>
+    [RegisterComponent]
+    [Access(typeof(GhostRoleSystem))]
+    public sealed class GhostRoleMobSpawnerComponent : GhostRoleSpawnerComponent
+    {
         [ViewVariables(VVAccess.ReadWrite)]
         [DataField("prototype", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
         public string? Prototype { get; private set; }
     }
+
+    /// <summary>
+    ///   Allows a ghost to take this role, spawning a randomized Humanoid
+    /// </summary>
+    [RegisterComponent]
+    [Access(typeof(GhostRoleSystem))]
+    public sealed class GhostRoleRandomSpawnerComponent : GhostRoleSpawnerComponent
+    {
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("settings", customTypeSerializer: typeof(PrototypeIdSerializer<RandomHumanoidSettingsPrototype>))]
+        public string Settings = default!;
+    }
+
 }
```

Comment:
I'm skeptical of having 4 non-abstract ghost role components, this should just be using the spawner itself and not relying on another component for it.